skbio/metadata/_mixin.py

```diff
@@ -406,3 +407,125 @@ def has_positional_metadata(self):
         """
         return (self._positional_metadata is not None and
                 len(self.positional_metadata.columns) > 0)
+
+
+class IntervalMetadataMixin(metaclass=abc.ABCMeta):
+    def __init__(self, interval_metadata=None):
+        raise NotImplementedError
+
+    def _init_(self, interval_metadata=None):
+        self.interval_metadata = interval_metadata
+
+    @property
+    @experimental(as_of="0.5.0-dev")
+    def interval_metadata(self):
+        '''``IntervalMetadata`` object containing info about interval features.
+
+        >>> from skbio import DNA
+        >>> from skbio.metadata import IntervalMetadata
+        >>> im = IntervalMetadata()
+        >>> intvl = im.add([(0, 3)], metadata={'gene': 'sagA'})
+        >>> seq = DNA('ACGT', interval_metadata=im)
+        >>> seq
+        DNA
+        --------------------------
+        Interval metadata:
+            1 interval features
+        Stats:
+            length: 4
+            has gaps: False
+            has degenerates: False
+            has definites: True
+            GC-content: 50.00%
+        --------------------------
+        0 ACGT
+        '''
+        if self._interval_metadata is None:
+            # Not using setter to avoid copy.
+            self._interval_metadata = IntervalMetadata()
+        return self._interval_metadata
+
+    @interval_metadata.setter
+    def interval_metadata(self, interval_metadata):
+        if interval_metadata is None:
+            interval_metadata = IntervalMetadata()
+        if isinstance(interval_metadata, IntervalMetadata):
+            # copy all the data to the mixin
+            self._interval_metadata = copy.deepcopy(interval_metadata)
```

Comment:
We want a shallow copy here instead of a deep copy to match the other metadata mixins.

`IntervalMetadata` doesn't define a `__copy__` method, which is what `copy.copy` calls (`copy.copy` performs a shallow copy). It *may* work as-is, but can you please add an `IntervalMetadata.__copy__` method with those semantics defined explicitly? I tend to not trust that custom classes will work with `copy.copy` (for example, numpy arrays and pandas dataframes don't always work with `copy.copy`, though that may have been fixed in more recent versions).

Also, above this line is where we'd want to validate that the supplied `interval_metadata` object's length matches `self._interval_metadata_axis_len_()`. This validation is similar to the positional metadata setter.